### PR TITLE
Remove notebook editor decoration type

### LINKED ISSUE #146838: Remove the notebookEditorDecorationType api proposal
The `notebookEditorDecorationType` currently is not on track for finalization. In order to decide what to do with it, we would like to know how this api is currently being used and if there would be any big concerns around removing it

Here are the current consumers:

- [x] Jupyter @rchiodo @DonJayamanne 
- [x] Jupyter powertoys @rchiodo @DonJayamanne 
- [x] Dot Net Interactive @brettfo, @colombod

Please check if your extension is using this api. If it is, can you explain how the api is used today and if there are alternative approach you could use to implement the functionality?

### PR DESCRIPTION
**Pull request by mjbvz**
Fixes #146838

### CODE REVIEW

**Comment by mjbvz** [comment]
@rebornix I split this into two commits: the first removes the API proposal and the second fully removes all the code that implemented it. Let me know if you feel we should keep some of this instead

Waiting until the latest Jupyter release goes out before merging this

**Review by rebornix — APPROVED** [review]

**Pull request merged**
